src/tags-input.js

```diff
@@ -53,12 +59,17 @@ tagsInput.directive('tagsInput', function($timeout, $document, tagsInputConfig,
 
         tagIsValid = function(tag) {
             var tagText = getTagText(tag);
+            var valid = tagText &&
+                     tagText.length >= options.minLength &&
+                     tagText.length <= options.maxLength &&
+                     options.allowedTagsPattern.test(tagText) &&
+                     !tiUtil.findInObjectArray(self.items, tag, options.displayProperty);
+
+            if (scope.onTagAdding()) {
+                valid = valid && scope.onTagAdding()(tag);
```

Comment:
The reason I had to invoke it via `scope.onTagAdding()(tag)` has to do with the callback being optional. If I change it to `scope.onTagAdding({ $tag: tag })` but the user didn't define this callback, the validation for `tagIsValid` will return `false` because `onTagAdding` returns `undefined`.

However, if the reference to the function can be passed through without having to invoke it, the `tagIsValid` method can check to make sure the callback was defined before invoking it.

Example:

    // Either undefined or a function will be returned from this
    onTagAddingCallback = scope.onTagAdding()

    if (onTagAddingCallback) { 
        okayToAdd = onTagAddingCallback(tag); 
    }

This can be demonstrated by changing it to `scope.onTagAdding({ $tag: tag })` and then run the tests. Every spec that does not define 'on-tag-adding' will fail, because `tagIsValid` ends up being `false`.

I can make the change you're requesting, but I'll have to find some other way to make sure that option is defined before considering it in the `tagIsValid` method.